corehq/apps/enterprise/enterprise.py

```diff
@@ -561,3 +564,36 @@ def rows_for_domain(self, domain_obj):
         num_errors = sum([result['direction_count'] for result in results if result['error']])
 
         return [(domain_obj.name, num_sent, num_received, num_errors), ]
+
+
+class EnterpriseAPIReport(EnterpriseReport):
+    title = gettext_lazy('API Usage')
+    total_description = gettext_lazy('# of Active API Keys')
+
+    @property
+    def headers(self):
+        return [_('Web User'), _('API Key Name'), _('Scope'), _('Expiration Date [UTC]'), _('Created On [UTC]'),
+                _('Last Used On [UTC]')]
+
+    @property
+    def rows(self):
+        return [self._get_api_key_row(api_key) for api_key in self.unique_api_keys()]
+
+    @property
+    def total(self):
+        return self.unique_api_keys().count()
+
+    def unique_api_keys(self):
+        usernames = self.account.get_web_user_usernames()
+        unique_users = [User.objects.get(username=username) for username in usernames]
+        return HQApiKey.objects.filter(user__in=unique_users, is_active=True)
```

Comment:
Three things we might want to look into here:
1. Can we simplify what is pulled back from the database -- i.e. instead of fetching the entire user object, can we fetch just the user ID or username
2. Can/should this be turned into a subquery so that we only send a single query to the database?
3. At what # of users are we concerned with `__in` performance? Maybe run this by the infrastructure team.